UI/Components/PreviousSegmentSettings.cs

```diff
@@ -202,6 +202,11 @@ private void rdoTimeSaveHundredths_CheckedChanged(object sender, EventArgs e)
 
         private void chkPossibleTimeSave_CheckedChanged(object sender, EventArgs e)
         {
+            if (!chkPossibleTimeSave.Checked && chkDropDecimalsPossibleTimeSave.Checked)
+            {
+                chkDropDecimalsPossibleTimeSave.Checked = false;
+                chkPossibleTimeSave.Checked = false;
```

Comment:
Why do you need this line if the `if` already makes sure this is `false`?

Reply:
It's weird, because if line 208 isn't added, then the code below that disables the fields doesn't actually disable the fields... it acts like `chkPossibleTimeSave.Checked` is `true` even though it goes into the if statement when it's `false`.

Reply:
I think it's because you mistakenly added `this.chkDropDecimalsPossibleTimeSave.CheckedChanged += new System.EventHandler(this.chkPossibleTimeSave_CheckedChanged);` in the designer code

Reply:
Tested by removing that line from designer and also the line on 208 and it doesn't seem like it changed the behavior described (ie. it's reaching line 207 in the `if` and disabling the drop decimals checkbox, but still not behaving as if `chkPossibleTimeSave.Checked` is `false` and disabling the fields, which it should be as it's going into the `if`).

Reply:
Not really sure why that would be happening - I would suggest debugging further to identify the root cause.